src/Sentry/Internal/SettingLocator.cs

```diff
@@ -35,19 +35,16 @@ public string GetDsn()
         // For DSN only
         // An empty string set on the option should NOT be converted to null because it is used
         // to indicate the the SDK is disabled.
-
-        var dsn = _options.Dsn;
-        if (dsn != null)
+        _options.Dsn ??= GetEnvironmentVariable(Constants.DsnEnvironmentVariable)
+                         ?? AssemblyForAttributes?.GetCustomAttribute<DsnAttribute>()?.Dsn;
+        if (_options.Dsn is null)
```

Comment:
Do we always call the locator after any framework bootstrapping phase? Like ASP.NET Core Configuration system? I imagine integration tests would break otherwise but worth testing a sample that loads the DSN from appsettings.json

Reply:
Yeah I think it's all covered by the integration tests. 

Some manual smoke tests on my part:

### ASP.NET Core

#### Configure Disabled DSN as Assembly Attribute and removing from Options Callback

Setting this in the `Directory.Build.props` in the Samples directory and removing any other DSN configuration in C# resulted in a DisabledHub (as expected):

```
  <ItemGroup>
    <SentryAttributes Include="Sentry.DsnAttribute">
      <_Parameter1></_Parameter1>
    </SentryAttributes>
  </ItemGroup>
```

#### Removing DSN as Assembly Attribute and from Options Callback

Removing the Assembly Attribute from the `Directory.Build.props` in the Samples directory and removing any other DSN configuration in C# code resulted in an ArgumentNullException, as expected:

```
/usr/local/share/dotnet/dotnet /Users/jamescrosswell/code/sentry-dotnet/samples/Sentry.Samples.AspNetCore.Basic/bin/Debug/net6.0/Sentry.Samples.AspNetCore.Basic.dll
  Debug: Logging enabled with ConsoleDiagnosticLogger and min level: Debug
Unhandled exception. System.ArgumentNullException: Value cannot be null. (Parameter 'You must supply a DSN to use Sentry.To disable Sentry, pass an empty string: "".See https://docs.sentry.io/platforms/dotnet/configuration/options/#dsn')
   at Sentry.Internal.SettingLocator.GetDsn() in /Users/jamescrosswell/code/sentry-dotnet/src/Sentry/Internal/SettingLocator.cs:line 42

```